Initial version base on Hamid's script.

kamil-holubicki opened the pull request.

kamil-holubicki committed:
Initial version base on Hamid's script.

edux commented on call-home.sh:
```diff
@@ -0,0 +1,286 @@
+
+# Script variables
+# Can be passed to the script via env variable or via named parameter
+PERCONA_INSTANCE_ID_FILE_PATH="${PERCONA_TELEMETRY_URL:-"/tmp/telemetry_uuid"}"
```
If telemetry_uuid should persist, please consider a different location as tmp will be cleaned by systemd-tmpfiles once per day or at shutdown.

kamil-holubicki replied:
done

maxkondr commented on README:
```diff
@@ -0,0 +1,30 @@
+usage: ./call-home.sh OPTIONS
+
+Collects telemetry information and sends it to a Percona server. The data collection
+may be disabled via setting an environment variable [PERCONA_TELEMETRY_DISABLE].
```
```suggestion
may be disabled via setting an environment variable [PERCONA_TELEMETRY_DISABLE=1].
```

hqakhtar replied:
Let's probably keep it simple. If the variable is set, we don't collect any data. If not, we do.

kamil-holubicki replied:
It will be not clear if it still works if `PERCONA_TELEMETRY_DISABLE=0`, so as it is implemented now is in my opinion OK.

maxkondr replied:
I, as user, prefer to get an exact and clear instructions what and how shall be passed to script, program, etc in order to make it do smth.

maxkondr commented on README:
```diff
@@ -0,0 +1,30 @@
+usage: ./call-home.sh OPTIONS
+
+Collects telemetry information and sends it to a Percona server. The data collection
```
```suggestion
Collects telemetry information and sends it to a Percona Telemetry service. The data collection
```

kamil-holubicki replied:
done

maxkondr commented on call-home.sh:
```diff
@@ -0,0 +1,286 @@
+
+usage: $0 OPTIONS
+
+Collects telemetry information and sends it to a Percona server. The data collection
```
```suggestion
Collects telemetry information and sends it to a Percona Telemetry service. The data collection
```

kamil-holubicki replied:
Done

maxkondr commented on call-home.sh:
```diff
@@ -0,0 +1,286 @@
+usage: $0 OPTIONS
+
+Collects telemetry information and sends it to a Percona server. The data collection
+may be disabled via setting an environment variable [PERCONA_TELEMETRY_DISABLE].
```
```suggestion
may be disabled via setting an environment variable [PERCONA_TELEMETRY_DISABLE=1].
```

kamil-holubicki replied:
done

maxkondr commented on call-home.sh:
```diff
@@ -0,0 +1,286 @@
+
+    if [[ -z "$PERCONA_PRODUCT_FAMILY" ]];
+    then
+        printf "PERCONA_PRODUCT_FAMILY not provided. %s\n" "$USAGE_TEXT" >&2
```
```suggestion
        printf "PERCONA_PRODUCT_FAMILY is not provided. %s\n" "$USAGE_TEXT" >&2
```

and for the rest as well

kamil-holubicki replied:
done

maxkondr commented on call-home.sh:
```diff
@@ -0,0 +1,286 @@
+collect_data_for_report()
+{
+    json_message_map["id"]=$(cat /proc/sys/kernel/random/uuid)
+    json_message_map["createTime"]="$(date +"%Y-%m-%dT%H:%M:%S%:z")"
```
actually we can't guarantee that user's time zone is UTC.

> date +"%Y-%m-%dT%H:%M:%S%:z"
> 2023-09-19T07:36:27+00:00

> date --iso-8601='seconds'
> 2023-09-19T07:36:37+03:00

hqakhtar replied:
Actually, we can change the date command to force output in UTC timezone.

date -u '+%Y-%m-%d %H:%M %Z'

kamil-holubicki replied:
```
$ date +"%Y-%m-%dT%H:%M:%S%:z"
2023-09-19T13:45:54+02:00
$ date --iso-8601='seconds'
2023-09-19T13:46:06+02:00

$ date -u '+%Y-%m-%d %H:%M %Z'
2023-09-19 11:46 UTC
```
So 1st and 2nd ones provide the same output. Should we use `date --iso-8601='seconds'`

@EngineeredVirus , 3rd option: Service API will reject it

maxkondr replied:
ok, up to implementor

maxkondr commented on call-home.sh:
```diff
@@ -0,0 +1,286 @@
+    case $OPTION in
+        h)
+            usage
+            exit_script 1
```
in case the script content will be used as is in package post-install - do not forget to change all exit codes to 0

kamil-holubicki replied:
Let's wait for Evgeniy's opinion

kamil-holubicki committed:
Addressed review comments.

Handling of escape characters done for the whole JSON string being sent.

kamil-holubicki committed:
1. Set the default telemetry config file path to /usr/local/percona/telemetry_uuid
2. Enhanced the logic around detection if the script already reported particular product

maxkondr commented on README:
```diff
@@ -0,0 +1,30 @@
+
+  -h  Show this message
+
+  -f  [PERCONA_PRODUCT_FAMILY]              Product family identifier.                                  [REQUIRED]
```
please add possible values for this param

maxkondr replied:
Possible values:
```
PRODUCT_FAMILY_MYSQL
  PRODUCT_FAMILY_MONGODB
  PRODUCT_FAMILY_POSTGRESQL
  PRODUCT_FAMILY_PMM
  PRODUCT_FAMILY_EVEREST
  PRODUCT_FAMILY_PERCONA_TOOLKIT
```

kamil-holubicki replied:
done

kamil-holubicki committed:
Changed the default telemetry reporting endpoint to Percona dev.

kamil-holubicki committed:
Added the info to the help section about possible product families.

eleo007 commented on call-home.sh:
```diff
@@ -0,0 +1,326 @@
+    json_message_map["product_family"]=$PERCONA_PRODUCT_FAMILY
+
+    metrics_map["pillar_version"]=$PERCONA_PRODUCT_VERSION
+    metrics_map["OS"]=${PERCONA_OPERATING_SYSTEM}
```
In the list of basic metrics for phase 0 there is HW architecture (at least from https://www.notion.so/percona/Pillars-product-usage-tracking-3a933f1e51404cd68e6e40691912556f ):
- operating system name and version (uname -ro or equivalent)
*** - hardware architecture (uname -m equivalent)
- pillar software installed
- pillar software version including hotfix number

Will we include it to OS?

kamil-holubicki replied:
Good point! Added as `hardware_arch`

kamil-holubicki committed:
Hardware architecture info added to collected metrics.

maxkondr commented on README:
```diff
@@ -0,0 +1,43 @@
+  -u  [PERCONA_TELEMETRY_URL]               Percona Telemetry Service endpoint                          [Default: https://check-dev.percona.com/v1/telemetry/GenericReport]
+  -t  [PERCONA_SEND_TIMEOUT]                Default timeout for the curl command.                       [Default: 10]
+
+Note that -d PERCONA_PRODUCT_FAMILY can be set to any string, but only the following ones will be accepted
```
```suggestion
Note that -f PERCONA_PRODUCT_FAMILY can be set to any string, but only the following ones will be accepted
```

kamil-holubicki replied:
fixed

kamil-holubicki committed:
Removed 'i' option from 'uname' call as it is not supported by Mac

kamil-holubicki committed:
Param typo fixed.

kamil-holubicki commented:
Moved the script to Percona-Lab/telemetry-agent.
New PR: https://github.com/Percona-Lab/telemetry-agent/pull/1
